### DISCUSSION TITLE
Is there a guideline how to update a semver version when increasing the minimal C++ level from C++14 to C++17

### DISCUSSION THREAD

**Discussion by vbaderks** [microsoft/vcpkg#28640] [Q&A]
I have a package on vcpkg that requires currently a C++14 compiler or newer to build the library and to use its header files.
I am planning to use C++17 for the next version and drop support for C++14. Both for building the library and using it header files.
The package is currently using a semver version number schema.

Are the guidelines\best practices for this case: do I need to update the major version (aka 2.4.0 -> 3,0.0)?

Would it still be possible to release a security fix for the 2.x.y branch after releasing 3.0.0 (for example 2.4.1)?

**Comment by Osyotr** [comment]
> Are the guidelines\best practices for this case: do I need to update the major version (aka 2.4.0 -> 3.0.0)?

Yes, as this is an incompatible change (https://semver.org/)

> Would it still be possible to release a security fix for the 2.x.y branch after releasing 3.0.0 (for example 2.4.1)?

No, vcpkg does not support that. (But you can release next version as a separate port, like it was done with opencv4 and opencv5)
